server/api/reminders_api.js

```diff
@@ -0,0 +1,169 @@
+        assetId: {
+        required: true,
+        type: "integer"
+        },
+        repeat: {
+        type: "string",
+        defaultValue: 'Yes'                               
+        },
+        timeZone: {
+        type: "string",
+        defaultValue: 'Eastern'                               
+        }
+});
+
+const UPDATE_VALIDATOR = new ApiValidator({                                        
+        assetId: {
+        required: true,
+        type: "integer"
+        },
+        deliverAt: {
+        type: "date",                              
+        },
+        timeZone: {
+        type: "string",
+        defaultValue: 'Eastern'                               
+        }
+});
+var router = require("express").Router();
+
+
+
+// Create a reminder.
+router.post("/", function(req, res) {
+  res.jsonResultOf(new Promise(function(resolve) {
+  
+    if (!req.user) {
+      throw { status: 401 };
+    }
+
+      
+    var fields = VALIDATOR.validateNew(req.body);
+
+
+    console.log('in the lc scheduler function' + fields.dateStr);
+    var vid = fields.vid ;
+    
+    var inputStr = fields.dateStr;
+    var date = new Date(inputStr);
+    var rightNow = new Date(Date.now() + 5000);
+    var newYorkTime = Moment.tz(date, "America/New_York");
+    var losAngeles = newYorkTime.clone().tz("America/Los_Angeles");
+    var centralAmerica = newYorkTime.clone().tz("America/Chicago");
+    var india = newYorkTime.clone().tz("Asia/Kolkata");
+
+    console.log('this is the date and time now::' + rightNow + "  and this was selected::" + date + "this is the timeZone:" + fields.timeZone );
+    console.log('New York :' + newYorkTime.format() + " Los Angeles :" + losAngeles.format() + " Central timeZone" + centralAmerica.format());
+
+    var deliverAt = newYorkTime.format() ; // defaulting to eastern timezone
+
+    if (fields.timeZone == "Eastern") {deliverAt = newYorkTime.format()} ;
+    if (fields.timeZone == "Central") {deliverAt = centralAmerica.format()} ;
+    if (fields.timeZone == "Pacific") {deliverAt = losAngeles.format()} ;
+    if (fields.timeZone == "IST") {deliverAt = india.format()} ;
+      
+    Reminders.create({
+                assetId: fields.assetId,
+                fromUserId: fields.fromUserId,
+                toUserId : fields.toUserId,
+                status: 1,
+                timeZone: fields.timeZone, 
+                Repeat : fields.repeat,
+                Expired : 'No', 
+                deliverAt: deliverAt,
+                lastDeliveredAt: 'Never'
+            }).then(function (Reminders) {
+                if(Reminders) {
+                console.log('new reminder created! ');
+                 res.status(200).send(Reminders);
+                }       
+            }).catch(function (err){
+                console.log('could not create new Reminder!');
+                res.status(500).send('Could not add the reminder, please contact admin ErroCode:' + err.parent.code);   
+            });

```

Comment:
Remove these then() and catch() blocks - they are redundant.  the res.jsonResultOf() call above already handles response formatting, in both success and error cases, but you are never resolving the promise passed to it.   To do that, simply: 

resolve(Reminders.create({...});